Source/WebKit/NetworkProcess/storage/LocalStorageManager.cpp

```diff
@@ -243,4 +243,9 @@ void LocalStorageManager::disconnectFromStorageArea(IPC::Connection::UniqueID co
         connectionClosedForTransientStorageArea(connection);
 }
 
+HashMap<String, String> LocalStorageManager::fetchStorageMap() const
+{
+    return m_localStorageArea->allItems();
```

Comment:
The LocalStorage items will be in either `m_localStorageArea` (first-party) **or** 
`m_transientStorageArea` (third-party), so you should null check `m_localStorageArea` and try getting from `m_transientStorageArea ` if `m_localStorageArea` is null.
And `m_transientStorageArea` is also pointer, so you need null check for `m_transientStorageArea` too.

To ensure the implementation is correct, please add a test for localStorage in third-party iframe. For how to add 3rd frame for API test, you might check FileSystemAccess.FetchDataForThirdParty.

Reply:
Added!